libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -483,8 +499,13 @@ export class PackageExtractor {
           path.join(packageJsonRealFolderPath, 'package.json')
         );
 
+        const targetSubspace: IExtractorSubspace | undefined = subspaces?.find(
+          (p) => p.pnpmInstallFolder && Path.isUnder(packageJsonFolderPath, p.pnpmInstallFolder)
+        );
+
         // Transform packageJson using the provided transformer, if requested
-        const packageJson: IPackageJson = transformPackageJson?.(originalPackageJson) ?? originalPackageJson;
+        const packageJson: IPackageJson =
+          targetSubspace?.transformPackageJson?.(originalPackageJson) ?? originalPackageJson;
```

Comment:
This API design is confusing for users because it ignores mistakes.  In the below example, what is the `pnpmInstallFolder` for the `default` subspace? It is unclear.

```ts
this.extractAsync({
  pnpmInstallFolder: 'abc',
  subspaces: [
    { subspaceName: 'default', pnpmInstallFolder: 'def' },
    { subspaceName:'other', pnpmInstallFolder: 'ghi' }
  ]
})
```

Reply:
(I'm working on a fix)

Reply:
https://github.com/EscapeB/rushstack/pull/1

Reply:
_Fixed_